### PR TITLE
refactor: use 'std::mem::swap' instead of for loop

### PR DESCRIPTION
**Pull request by woshilapin**
Trying to simplify the code. What is really done in this piece of code is:
- exchanging the values of the new `debark_front` into the current one
- clearing out the new `debark_front` for the next iteration of the engine

I believe the proposed modification makes more explicit the intent. It might also be more performant because it avoids an iteration, some cloning and some functions calls.

### CODE REVIEW

**Review by xlqian — APPROVED** [review]
:+1:

File: src/engine/multicriteria_raptor.rs [hunk 1]
```diff
@@ -489,19 +489,17 @@ where
...
-                debark_front.add_unchecked(*debark, criteria.clone());
-            }
-            new_debark_front.clear();
+            // we do not need to check, because
+            //  - new_debarked_front is a pareto front
+            //  - we added an element to new_debarked_front only if it was not dominated by debarked_front
+            //  - we removed from debarked_front all elements that were dominated by an element of new_debarked_front
+            //
+            // TODO : add debug_assert here to check what is written above
+            std::mem::swap(
```
Comment (xlqian) [inline src/engine/multicriteria_raptor.rs:498]: so here, a simple swap also implies that debark_front is empty before the swap, right?

**Comment by pbench** [comment]
> Trying to simplify the code. What is really done in this piece of code is:
> 
>     * exchanging the values of the new `debark_front` into the current one
> 
>     * clearing out the new `debark_front` for the next iteration of the engine

There is some misunderstanding here : the code **does not** perform a swap between `new_debark_front` and `debark_front`.
Rather, it adds all elements of `new_debark_front` into `debark_front`, and then clear `new_debark_front`. But `debark_front` is not cleared. 
Here:
 - `new_debark_front` corresponds to the new journeys explored in the last round
 - `debark_front` corresponds to the journeys explored in all rounds before the last one
So this function is cleaning things up before we start the next round.

**Comment by woshilapin** [comment]
I don't think I'll take any time on this soon and it's not super important. The important part is that the code I've made is not valid, adding test would only be a nice bonus. I'm closing this PR.

**Pull request closed without merging**